OpenRA.Mods.Common/Widgets/Logic/ConnectionLogic.cs

```diff
@@ -159,13 +157,14 @@ public ConnectionSwitchModLogic(Widget widget, OrderManager orderManager, Action
 			var abortButton = panel.Get<ButtonWidget>("ABORT_BUTTON");
 			var switchButton = panel.Get<ButtonWidget>("SWITCH_BUTTON");
 
+			var modId = orderManager.ServerExternalMod.Id;
 			var modTitle = orderManager.ServerExternalMod.Title;
 			var modVersion = orderManager.ServerExternalMod.Version;
 			var modIcon = orderManager.ServerExternalMod.Icon;
 
 			switchButton.OnClick = () =>
 			{
-				var launchCommand = "Launch.Connect=" + orderManager.Host + ":" + orderManager.Port;
+				var launchCommand = "Launch.URI={0}".F(new UriBuilder("openra-" + modId + "-" + modVersion, orderManager.Connection.EndPoint.Address.ToString(), orderManager.Connection.EndPoint.Port));
```

Comment:
I'd rather not hardcode the mod uri scheme here - this line will probably be forgotten if we want to change it in the future. Can we use `tcp://` like we do in `LaunchArguments` instead?